fedotov_yaroslav/graph.cpp

```diff
@@ -0,0 +1,76 @@
+#include <vector>
+
+class Graph {
+  using VertexId = int;
+  using EdgeId = int;
+
+ public:
+  void add_vertex() { vertices_.emplace_back(last_vertex_id_++); }
+
+  void add_edge(VertexId first, VertexId second) {
+    edges_.emplace_back(last_edge_id_++,first, second);
```

Comment:
Похоже что у тебя тут баг.
Добавь приватный метод 
```cpp
bool has_vertex(VertexId id) const;
```
И добавь внутрь текущего метода проверки на то что вершины существуют:
```suggestion
  void add_edge(VertexId first, VertexId second) {
    assert(has_vertex(first));
    assert(has_vertex(second));
    edges_.emplace_back(last_edge_id_++, first, second);
```